### ISSUE TITLE
Type forward references do not work with PEP 585 collections

### ISSUE THREAD

**Issue by intgr** [oxan/djangorestframework-dataclasses#51]
I don't understand the reason for this yet, but djangorestframework-dataclasses currently fails when using forward references with PEP 585 collection syntax, for example `list["SomeClass"]`.

It works, however, when using `typing.List["SomeClass"]` or `list[SomeClass]` (without a forward reference).

I wrote this test case to reproduce the issue:

```python
@dataclasses.dataclass
class Simple:
    value: str


class IssuesTest(TestCase):
    # Issue #51: Type forward references do not work with PEP 585 collections
    def test_forward_reference_list(self):
        @dataclasses.dataclass
        class WithForwardReference:
            """Use quoted type hints (e.g. forward references from PEP 484)"""
            child: 'Simple'
            # children: typing.List['Simple']   <-- works when uncommenting this
            children: list['Simple']

        serializer = DataclassSerializer(
            dataclass=WithForwardReference, data={'child': {'value': 'test1'}, 'children': [{'value': 'test2'}]}
        )
        serializer.is_valid(raise_exception=True)
        data: WithForwardReference = serializer.validated_data

        self.assertEqual('test1', data.child.value)
        self.assertEqual('test2', data.children[0].value)
```

The result is:
> NotImplementedError: Automatic serializer field deduction not supported for field 'children' on 'WithForwardReference' of type 'list['Simple']' (during search for field of type 'Simple').

**Comment by oxan** [comment]
I bet `typing.get_type_hints()` doesn't resolve such forward references.

**Comment by intgr** [comment]
WTF, this is a known Python bug since 2020: https://bugs.python.org/issue41370

**Comment by oxan** [comment]
This is fixed by #53 now, thanks for your detailed report!

**Comment by intgr** [comment]
Thanks! I considered opening a PR for this, but didn't find time for it yet.
